app/models/setting.rb

```diff
@@ -268,4 +270,8 @@ def refresh_registry_value
       definition.value_from_db = value
     end
   end
+
+  def remove_whitespaces
+    self[:value] = value.each { |a| a.strip! if a.respond_to? :strip! }
```

Comment:
I suggest using `is_a?(String)` to check if `a` is a string, which is a more precise check than `respond_to?(:strip!)`. This avoids calling `strip!` on non-string objects, which could raise an error.

```suggestion
    self[:value] = value.each { |a| a.is_a?(String) ? a.strip! : a }
```